plugins/out_s3/s3.c

```diff
@@ -652,13 +674,118 @@ static int cb_s3_init(struct flb_output_instance *ins,
+                          DEFAULT_PARQUET_COMPRESSION_FORMAT_UPCASES);
+        }
+        else {
+            if (strncasecmp(tmp, "uncompressed", 12) == 0 ||
+                strncasecmp(tmp, "snappy", 6) == 0 ||
+                strncasecmp(tmp, "gzip", 4) == 0 ||
+                strncasecmp(tmp, "zstd", 4) == 0) {
+                flb_plg_info(ctx->ins, "parquet.compression format is %s", tmp);
+            }
+            else if (strncasecmp(tmp, "lzo", 3) == 0 ||
+                strncasecmp(tmp, "brotli", 6) == 0 ||
+                strncasecmp(tmp, "lz4", 3) == 0) {
+                flb_plg_info(ctx->ins, "unsupported parquet.compression format %s", tmp);
+            }
+            else {
+                flb_plg_error(ctx->ins, "unknown parquet.compression format %s", tmp);
+                return -1;
+            }
+            for (i = 0; i < strlen(tmp) || i < sizeof(buf); i++) {
+                buf[i] = toupper(tmp[i]);
+            }
+
+            ctx->parquet_compression = flb_sds_create_len(buf, strlen(buf));
+        }
+
+        tmp = flb_output_get_property("parquet.record_type", ins);
+        if (!tmp) {
+            flb_plg_info(ctx->ins, "parquet.record_type format is %s",
+                         DEFAULT_PARQUET_RECORD_TYPE);
+            ctx->parquet_record_type = \
+                    flb_sds_create_len(DEFAULT_PARQUET_RECORD_TYPE,
+                                       strlen(DEFAULT_PARQUET_RECORD_TYPE));
+        }
+        else {
+            if (strncasecmp(tmp, "json", 4) == 0) {
+                tmp = "jsonl"; /* json should be interpreted as jsonl */
+                flb_plg_info(ctx->ins, "parquet.record_type format is %s", tmp);
+            }
+            else if (strncasecmp(tmp, "msgpack", 7) == 0 ||
+                strncasecmp(tmp, "jsonl", 5) == 0) {
+                flb_plg_info(ctx->ins, "parquet.record_type format is %s", tmp);
+            }
+            else {
+                flb_plg_error(ctx->ins, "unknown parquet.record_type format %s", tmp);
+                return -1;
+            }
+            ctx->parquet_record_type = flb_sds_create_len(tmp, strlen(tmp));
+        }
+
+        tmp = flb_output_get_property("parquet.schema_type", ins);
+        if (!tmp) {
+            flb_plg_info(ctx->ins, "parquet.schema_type format is %s",
+                         DEFAULT_PARQUET_SCHEMA_TYPE);
+            ctx->parquet_schema_type = \
+                    flb_sds_create_len(DEFAULT_PARQUET_SCHEMA_TYPE,
+                                       strlen(DEFAULT_PARQUET_SCHEMA_TYPE));
+        }
+        else {
+            if (strncasecmp(tmp, "msgpack", 7) == 0 ||
+                strncasecmp(tmp, "json", 4) == 0) {
+                flb_plg_info(ctx->ins, "parquet.record_type format is %s", tmp);
+            }
+            else {
+                flb_plg_error(ctx->ins, "unknown parquet.record_type format %s", tmp);
+                return -1;
+            }
```

Comment:
Seems like the logic and logs here should be around parsing `parquet.schema_type` not `parquet.record_type`?

Reply:
Yeah, shouldn't this be testing for `avro` and `bigquery`?

As per:
```
    {
     FLB_CONFIG_MAP_STR, "parquet.schema_type", "avro",
     0, FLB_FALSE, 0,
    "Record type for parquet objects. 'avro' and 'bigquery' are the supported values. "
    "Defaults to avro. "
    },
```

Reply:
Yes. It's correct. Thanks for pointing this out.